infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/LRRollingUpgradeHandler.java

```diff
@@ -81,5 +90,29 @@ public boolean isLRUpgradeInProgress(TxnContext txnContext) {
      */
     public void migrateData(TxnContext txnContext) {
         // Data migration to be added here.
+        eventSnapshotSyncRollingUpgrade(txnContext);
```

Comment:
we are now introducing the behavior that a snapshot sync gets triggered on every restart.  Let us discuss how we can avoid it with this change.